airbyte-integrations/connectors/source-coinmarketcap/source_coinmarketcap/coinmarketcap.yaml

```diff
@@ -0,0 +1,86 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: ["data"]
+  requester:
+    url_base: "https://pro-api.coinmarketcap.com"
+    http_method: "GET"
+    authenticator:
+      type: ApiKeyAuthenticator
+      header: "X-CMC_PRO_API_KEY"
+      api_token: "{{ config['api_key'] }}"
+  offset_paginator:
+    type: DefaultPaginator
+    $options:
+      url_base: "*ref(definitions.requester.url_base)"
+    page_size_option:
+      inject_into: "request_parameter"
+      field_name: "limit"
+    pagination_strategy:
+      type: "OffsetIncrement"
+      page_size: 1000
+    page_token_option:
+      field_name: "start"
+      inject_into: "request_parameter"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      $ref: "*ref(definitions.offset_paginator)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  categories_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "categories"
+      primary_key: "id"
+      path: "/v1/cryptocurrency/categories"
+  listing_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "listing"
+      primary_key: "id"
+      path: "/v1/cryptocurrency/listings/{{ config['data_type'] or ''}}"
```

Comment:
Nice idea, but unfortunately it will not work here, because i'am using a free account to test, and  /historical endpoint require a premium account, that will block the acceptance test.

Reply:
okay. that makes sense. Thanks for clarifying.